packages/components/src/tools-panel/tools-panel-header/component.js

```diff
@@ -27,35 +41,87 @@ const ToolsPanelHeader = ( props, forwardedRef ) => {
 		return null;
 	}
 
+	const defaultItems = Object.entries( menuItems?.default || {} );
+	const optionalItems = Object.entries( menuItems?.optional || {} );
+
 	return (
 		<h2 { ...headerProps } ref={ forwardedRef }>
 			{ labelText }
 			{ hasMenuItems && (
-				<DropdownMenu icon={ moreVertical } label={ labelText }>
+				<DropdownMenu
+					icon={ moreVertical }
+					label={ labelText }
+					menuProps={ { style: { minWidth: '200px' } } }
+				>
 					{ ( { onClose } ) => (
 						<>
-							<MenuGroup label={ __( 'Display options' ) }>
-								{ Object.entries( menuItems ).map(
-									( [ label, isSelected ] ) => {
-										return (
-											<MenuItem
-												key={ label }
-												icon={ isSelected && check }
-												isSelected={ isSelected }
-												onClick={ () => {
-													toggleItem( label );
-													onClose();
-												} }
-												role="menuitemcheckbox"
-											>
-												{ label }
-											</MenuItem>
-										);
-									}
-								) }
-							</MenuGroup>
+							{ !! defaultItems?.length && (
```

Comment:
This block is starting to get quite hard to parse, have you considered extracting it to a separate component? (it could live in this same file and it wouldn't need to be exported).

Also, have you considered [using ternaries instead of the `&&` operator](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx)?

Reply:
> This block is starting to get quite hard to parse, have you considered extracting it to a separate component? 

Agreed. I've wavered on extracting the `MenuItem`s to their own components previously but it didn't reduce the code much. I'll look at refactoring the menu groups out into separate components instead.

> Also, have you considered using ternaries instead of the && operator?

In this instance, the use of `!!` was to specifically guard against the issue highlighted by that link. The `&&` operator still appears to be the prevailing style within the codebase. I understand that using a ternary makes missing the `!!` a non-issue. Is this something that should be added to the Gutenberg coding guidelines?

Reply:
>  I'll look at refactoring the menu groups out into separate components instead.

Sounds good! Thank you :)

> In this instance, the use of !! was to specifically guard against the issue highlighted by that link. The && operator still appears to be the prevailing style within the codebase. I understand that using a ternary makes missing the !! a non-issue. Is this something that should be added to the Gutenberg coding guidelines?

Thank you for the explanation about using `!!`. I'm not sure if this should be added to the coding guidelines or even as an ESLint rule — that is something that @gziolo may advise with!